### PR TITLE
Minor changes to `src/chess`

### PR DESCRIPTION
**Pull request by ddobbelaere**
Some minor changes to `src/chess/board.{h,cc}`

If I am not mistaken, `ChessBoard::GenerateLegalMovesAndPositions` is not used anymore? Can this be removed or is it still present to cross-check some possible future optimizations of `IsLegalMove` used in `ChessBoard::GenerateLegalMoves`?

### CODE REVIEW

**Comment by ddobbelaere** [comment]
Added some hash calculation improvements. Yields small speedup (~0.5%) for default params. Also fixes poor "spreading" of `positions` (that at the moment only seems necessary to allow for future dynamic parameter changes?).

File: src/chess/board.h [hunk 1]
```diff
@@ -81,8 +81,10 @@ class ChessBoard {
   uint64_t Hash() const {
     return HashCat({our_pieces_.as_int(), their_pieces_.as_int(),
                     rooks_.as_int(), bishops_.as_int(), pawns_.as_int(),
-                    our_king_.as_int(), their_king_.as_int(),
-                    castlings_.as_int(), flipped_});
+                    (static_cast<unsigned>(our_king_.as_int()) << 24) |
```
Comment (mooskagh) [inline src/chess/board.h]: `unsigned` has different length at different platforms, it's safer to cast to something of known size (e.g. `uint64_t` or `uint32_t`)

**Comment by mooskagh** [comment]
`GenerateLegalMovesAndPositions` is from time when every node contained position. Can be removed now!

**Comment by ddobbelaere** [comment]
Reverted modifications to `PositionHistory::HashLast`. I think `uint64_t hash = positions;` is not critical at all (in fact, `uint64_t hash = 0;` would do also), as `positions` is implicitly influencing number of hashed positions.

File: src/chess/board.h (hunk 1 above)
Comment (ddobbelaere) [reply src/chess/board.h]: Thanks, you are right!

**Pull request merged**